test/test_sim_learn.py

```diff
@@ -39,4 +39,7 @@ def test_ip_mod2_1_n_bent(self):
     def test_ip_mod2_1_1_bent(self):
         sim_learn.main(["sim_learn", "16", "2", "1_1_bent", "ip_mod2", "12000", "1", "2", "1234", "1234"])
 
+    def test_permutation_atf(self):
+        sim_learn.main(["sim_learn", "16", "2", "permutation_atf", "xor", "12000", "1", "2", "1234", "1234"])
```

Comment:
please use small test size, for instance n=8 and N=10, to avoid long-running tests